### PR TITLE
Update README with Contributing Info

### PR DESCRIPTION
**Pull request by mgrauer**

### CODE REVIEW

**Comment by ronichoudhury** [comment]
If you also put this text in CONTRIBUTING.md (or, I think, .txt, .rst, etc.) it will show up as a link when someone goes to make a PR.

**Comment by mgrauer** [comment]
I moved this all over to a Contributing doc, which may not be exactly right, as that is what should show up when someone makes a PR.  I mainly meant this branch to get a starting point.  Perhaps I can move most of the info back to the README and only the PR stuff to the Contributing?

Related to the PRs, if anyone has any specific desires for formatting or requirements on PRs, this is a good place to speak up about it.  At this point there aren't so many that we can't manage them, and may not want to impose friction to contribute.  That could change as we grow.

@girder/developers

**Comment by zachmullen** [comment]
This LGTM

**Comment by mgrauer** [comment]
@zachmullen :runner: 

I added enough to this that I feel your LGTM is stale.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+When you submit a PR to the Girder repo, Travis CI will run the full build on two different branches
+
+  * The commit at the head of the PR branch, the `push` build
+  * The head of the PR branch that is then merged into `master`, the `pr` branch
+
+The Travis build will run according to the [.travis.yml file](/.travis.yml), which is useful as an example for how to set up your own environment for testing.
+
+#### Low Virtual Memory on Travis causing earlier termination of builds
+
+Sometimes Travis starts a build in a virtualization environment with a low amount of memory cache (we'd love to know why), which will cause the build to fail.  To catch this earlier, there is a [vmtest script](https://github.com/girder/girder/blob/master/scripts/vmtest.py) run in the [travis buildfile](https://github.com/girder/girder/blob/master/.travis.yml#L8) before the full test suite is run.  If this detects a low cache value, it aborts the build.  You'll notice this by an extremely short build log and a note in the log about low virtual memory.  If this happens, simply restart the Travis build.
```
Comment (zachmullen) [inline CONTRIBUTING.md]: @mgrauer As of today, we can delete the above paragraph! :tada:

**Comment by zachmullen** [comment]
This LGTM.

**Pull request merged**